sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -498,6 +498,9 @@ private[parquet] class ParquetRowConverter(
           int96RebaseSpec,
           wrappedUpdater)
 
+      case t: VariantType =>
```

Comment:
I'm surprised. Without this case match, why did variant ever work in parquet before?

Reply:
I think because the vectorized parquet reader is used by default, and I guess the vectorized reader doesn't use these converters. I had to set `PARQUET_VECTORIZED_READER_ENABLED` to false in the unit test to trigger the failure that this part of the change fixes.

Reply:
oh I see